java/src/test/java/ai/rapids/cudf/TableTest.java

```diff
@@ -2219,6 +2221,79 @@ void testGroupBySum() {
     }
   }
 
+  @Test
+  void testGroupByFirstExcludeNulls() {
+    try (Table t1 = new Table.TestBuilder()
```

Comment:
Other tests had issues with ordering for hash aggregate.  These require a sort aggregate so why not make the test simpler?
```
try (Table input = new Table.TestBuilder()
            .column(  1,   1,    1,  1,  2,    2,  2,    2)
            .column(null, 13, null, 12, 14, null, 15, null)
            .build();
       Table expected = new Table.TestBuilder()
            .column(0, 1)
            .build();
       Table found = input.groupBy(0).aggregate(first(1, false))) {
    assertTablesAreEqual(expected, found);
}
```

Reply:
Thanks. Not only simpler, but more thorough since this tests the validity buffer as well and seems to have exposed another issue. I've pushed the simplified tests.